src/components/collection/Selector/List/CollectionSelectorBody.tsx

```diff
@@ -0,0 +1,141 @@
+
+import { useRef } from 'react';
+
+import { Box, TableBody, TableCell, TableRow } from '@mui/material';
+
+import { useUnmount } from 'react-use';
+import AutoSizer from 'react-virtualized-auto-sizer';
+import { FixedSizeList } from 'react-window';
+
+import {
+    COLLECTION_SELECTOR_UUID_COL,
+    DEFAULT_ROW_HEIGHT,
+} from 'src/components/collection/Selector/List/shared';
+import { TABLE_BODY_CELL_CLASS_PREFIX } from 'src/components/tables/EntityTable/shared';
+import { useBinding_currentBindingUUID } from 'src/stores/Binding/hooks';
+
+function CollectionSelectorBody({
+    columns,
+    filterValue,
+    rows,
+    selectionEnabled,
+    setCurrentBinding,
+    tableScroller,
+    scrollingElementCallback,
+    checkScrollbarVisibility,
+}: CollectionSelectorBodyProps) {
+    const hackyTimeout = useRef<number | null>(null);
+    const scrollingElementRef = useRef<FixedSizeList | undefined>(undefined);
+    const virtualRows = useRef<any | null>(null);
+
+    const currentBindingUUID = useBinding_currentBindingUUID();
+
+    useUnmount(() => {
+        if (hackyTimeout.current) clearTimeout(hackyTimeout.current);
+    });
+
+    const handleCellClick = (id?: string) => {
+        if (
+            selectionEnabled &&
+            setCurrentBinding &&
+            id !== currentBindingUUID
+        ) {
+            // TODO (JSONForms) This is hacky but it works.
+            // It clears out the current binding before switching.
+            //  If a user is typing quickly in a form and then selects a
+            //  different binding VERY quickly it could cause the updates
+            //  to go into the wrong form.
+            setCurrentBinding(null);
+
+            if (typeof id === 'string') {
+                hackyTimeout.current = window.setTimeout(() => {
+                    setCurrentBinding(id);
+                });
+            }
+        }
+    };
+
+    return (
+        <TableBody component="div">
+            <AutoSizer>
+                {({ height, width }: AutoSizer['state']) => {
+                    return (
+                        <FixedSizeList
+                            ref={scrollingElementCallback}
+                            innerRef={virtualRows}
+                            outerRef={scrollingElementRef}
+                            height={height} // Adjust for header height
+                            itemCount={rows.length}
+                            itemData={rows}
+                            itemKey={(index, data) =>
+                                data[index][COLLECTION_SELECTOR_UUID_COL]
+                            }
+                            itemSize={DEFAULT_ROW_HEIGHT} // Row height
+                            overscanCount={10}
+                            width={width}
+                            onItemsRendered={() => {
+                                checkScrollbarVisibility();
+                            }}
+                        >
```

Comment:
This is a major change to take us from `DataGrid` to using a normal `Table`.

On item rendering checking the scrollbar helps with performance a _tiny_ bit.